### PR TITLE
Export classes from dart:collection.

### PR DESCRIPTION
**Pull request by lrhn**
In preparation of moving the classes to this package in Dart 2.0.

### CODE REVIEW

**Comment by floitschG** [comment]
LGTM.

**Comment by kevmoo** [comment]
You'll need to exclude dartfmt from stable, it seems.

Or only put it in `include`.

See https://github.com/dart-lang/test/blob/5750bd37e6642ef7854f7d9eee6c8581f0ca0993/.travis.yml#L60

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 1.15.0
+
+* Re-export a number of platform library classes from `dart:collection`
+  in anticipation of moving the classes to this package for Dart 2.0.
```
Comment (nex3) [inline CHANGELOG.md]: List the classes that are being re-exported, so users of those classes can search the changelog for a class name to see which version of the package first exposed that class.

File: lib/collection.dart [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
+    DoubleLinkedQueueEntry,
+    HasNextIterator,
+    LinkedList,
+    LinkedListEntry,
+    MapView, // TODO(lrn): Drop this class, it is equivalent to DelegatingMap.
+    SplayTreeMap,
+    SplayTreeSet,
+    UnmodifiableListView,
+    UnmodifiableMapBase,
+    UnmodifiableMapView;
```
Comment (nex3) [inline lib/collection.dart]: The unmodifiable classes are widely used. I think we should consider leaving them in `dart:collection`.

File: pubspec.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: collection
-version: 1.14.4-dev
+version: 1.15.0
```
Comment (kevmoo) [inline pubspec.yaml]: On 2nd thought, I'd hold off doing this. Copy-paste the classes and then deploy.

We really want an upper bound on the SDK constraint so folks don't get this package w/ the SDK that doesn't have these classes.

File: lib/collection.dart (hunk 2 above)
Comment (lrhn) [reply lib/collection.dart]: Being widly used isn't by itself enough to warrant a place in the platform libraries. We have `package:collection` (and friends) to hold generally useful code that doesn't *need* to be in the platform libraries.

File: pubspec.yaml (hunk 3 above)
Comment (lrhn) [reply pubspec.yaml]: There is an upper-bound on the SDK constraint of <2.0.0, and we will not remove the classes from Dart before 2.0.0. 

Is the problem that some 2.0.0-rcX will count as <2.0.0?

File: lib/collection.dart (hunk 2 above)
Comment (nex3) [reply lib/collection.dart]: These classes are [used occasionally](https://github.com/dart-lang/sdk/blob/cc5cb89e0b4ec788ad3b37d577f49ca7fa39b886/sdk/lib/io/platform_impl.dart#L110) in the SDK, but I'm also skeptical of the principle of stripping absolutely everything from the core libraries that's not strictly necessary for representing the transitive closure of the language spec and `dart:io`'s APIs or whatever.

The unmodifiable classes' APIs move in lock-step with the collection APIs, so they don't get any added flexibility from being in a package. They're not only widely-used, they're a fundamental part of exposing an API surface with strong encapsulation guarantees. I don't see the benefit of removing them from the core libraries, whereas requiring many users to depend on `collection` is a real cost.

**Comment by matanlurey** [comment]
Do we still want to move here @lrhn @kevmoo?

**Comment by lrhn** [comment]
Didn't happen for Dart 2. We'll reconsider for Dart 3.

**Pull request closed without merging**